tests/test__collection_api.py

```diff
@@ -5983,3 +5983,34 @@ def test_elem_match(self):
         ids = set(doc['_id'] for doc in self.db.collection.find(
             {'arr': {'$elemMatch': {'$lt': 10, '$gt': 4}}}, {'_id': 1}))
         self.assertEqual({1, 2}, ids)
+
+    def test_list_collection_names_filter(self):
+        self.db.create_collection('aggregator')
+        for day in range(10):
+            new_date = datetime.now() - timedelta(day)
+            self.db.create_collection('historical_{0}'.format(new_date.strftime('%Y_%m_%d')))
+
+        # test without filter
+        assert len(self.db.list_collection_names()) == 11
```

Comment:
We use self.assertEqual in this test suite to have nicer error messages. Please do so in your new test